plugins/inputs/slurm/slurm_test.go

```diff
@@ -0,0 +1,378 @@
+package slurm
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+	"github.com/stretchr/testify/require"
+)
+
+func TestURLs(t *testing.T) {
+	for _, url := range []string{"http://example.com:6820", "https://example.com:6820"} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.NoError(t, plugin.Init())
+	}
+
+	for _, url := range []string{"httpp://example.com:6820", "httpss://example.com:6820"} {
```

Comment:
If really want to test different problematic URL scenario's I would also add ones that will trigger the other errors, like empty URL or no hostname..

Reply:
Thanks a ton for the suggestion! I added those corner cases as well on c3414e6016d54e34739b235dda0953c4ace4613c.